File: Chapter 06/C#/Section_6_3_1_REST/FakeRestHandler.cs

using System.Net;

namespace Chapter06.RestExample
{
    // THE FAKE ENDPOINT
    // Intercepts any outbound HTTP call and returns our hardcoded JSON instead.
    public class FakeRestHandler : HttpMessageHandler
    {
        protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
        {
            // The exact JSON from Listing 6.1
            string jsonResponse = @"{
                ""id"": ""123"",
                ""name"": ""Salt & Vinegar Chips"",
                ""price"": 1.50,
                ""calories"": 250,
                ""ingredients"": [ ""Potatoes"", ""Oil"", ""Salt"" ],
                ""manufacturer"": { ""name"": ""SnackCorp"", ""address"": ""123 Food Lane"" }
            }";
            // Package it up as a standard HTTP 200 OK response
            var response = new HttpResponseMessage(HttpStatusCode.OK)
            {
                Content = new StringContent(jsonResponse, System.Text.Encoding.UTF8, "application/json")
            };

            return Task.FromResult(response);
        }
    }
}

File: Chapter 06/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter06/section_6_3_1_rest/FakeRestHandler.java
package com.grokkingsoftwarearchitecture.chapter06.section_6_3_1_rest;

/**
 * THE FAKE ENDPOINT.
 * Intercepts outbound HTTP calls to simulate a REST API response.
 * By encapsulating this here, we keep our test environment deterministic.
 */
public class FakeRestHandler {
    
    /**
     * Simulates an HTTP GET request to a remote server.
     *
     * @param url The mock endpoint being called.
     * @return A raw JSON string representing the full REST resource.
     */
    public String get(String url) {
        // The exact JSON payload, forcing the client to download everything
        return "{\n" +
               "  \"id\": \"123\",\n" +
               "  \"name\": \"Salt & Vinegar Chips\",\n" +
               "  \"price\": 1.50,\n" +
               "  \"calories\": 250,\n" +
               "  \"ingredients\": [ \"Potatoes\", \"Oil\", \"Salt\" ],\n" +
               "  \"manufacturer\": { \"name\": \"SnackCorp\", \"address\": \"123 Food Lane\" }\n" +
               "}";
    }
}